File: shared/athena-lms-common/src/main/java/com/athena/lms/common/event/EventTypes.java
package com.athena.lms.common.event;

/**
 * Canonical event type constants for all LMS domain events.
 * These strings are used as RabbitMQ routing keys.
 */
public final class EventTypes {

    private EventTypes() {}

    // ─── Account events ────────────────────────────────────────────────────────
    public static final String ACCOUNT_CREATED          = "account.created";
    public static final String ACCOUNT_CREDIT_RECEIVED  = "account.credit.received";
    public static final String ACCOUNT_DEBIT_PROCESSED  = "account.debit.processed";
    public static final String ACCOUNT_FROZEN           = "account.frozen";
    public static final String ACCOUNT_UNFROZEN         = "account.unfrozen";
    public static final String ACCOUNT_CLOSED           = "account.closed";

    // ─── Loan origination events ───────────────────────────────────────────────
    public static final String LOAN_APPLICATION_SUBMITTED  = "loan.application.submitted";
    public static final String LOAN_APPLICATION_APPROVED   = "loan.application.approved";
    public static final String LOAN_APPLICATION_REJECTED   = "loan.application.rejected";
    public static final String LOAN_DOCUMENTS_VERIFIED     = "loan.documents.verified";
    public static final String LOAN_CREDIT_ASSESSED        = "loan.credit.assessed";

    // ─── Loan management events ────────────────────────────────────────────────
    public static final String LOAN_DISBURSED          = "loan.disbursed";
    public static final String LOAN_REPAYMENT_RECEIVED = "loan.repayment.received";
    public static final String LOAN_DPD_UPDATED        = "loan.dpd.updated";
    public static final String LOAN_STAGE_CHANGED      = "loan.stage.changed";
    public static final String LOAN_CLOSED             = "loan.closed";
    public static final String LOAN_WRITTEN_OFF        = "loan.written.off";
    public static final String LOAN_MODIFIED           = "loan.modified";

    // ─── Payment events ────────────────────────────────────────────────────────
    public static final String PAYMENT_INITIATED   = "payment.initiated";
    public static final String PAYMENT_COMPLETED   = "payment.completed";
    public static final String PAYMENT_FAILED      = "payment.failed";
    public static final String PAYMENT_REVERSED    = "payment.reversed";

    // ─── Float events ──────────────────────────────────────────────────────────
    public static final String FLOAT_DRAWN              = "float.drawn";
    public static final String FLOAT_REPAID             = "float.repaid";
    public static final String FLOAT_FEE_CHARGED        = "float.fee.charged";
    public static final String FLOAT_RESTRICTION_APPLIED = "float.restriction.applied";
    public static final String FLOAT_LIMIT_CHANGED      = "float.limit.changed";

    // ─── AML / compliance events ───────────────────────────────────────────────
    public static final String AML_ALERT_RAISED    = "aml.alert.raised";
    public static final String AML_SAR_FILED       = "aml.sar.filed";
    public static final String CUSTOMER_KYC_PASSED = "customer.kyc.passed";
    public static final String CUSTOMER_KYC_FAILED = "customer.kyc.failed";
}

File: go-services/internal/common/event/event_types.go
package event

// Canonical event type constants for all LMS domain events.
// These strings are used as RabbitMQ routing keys.
// Must match Java EventTypes.java exactly.
const (
	// Account events
	AccountCreated        = "account.created"
	AccountCreditReceived = "account.credit.received"
	AccountDebitProcessed = "account.debit.processed"
	AccountFrozen         = "account.frozen"
	AccountUnfrozen       = "account.unfrozen"
	AccountClosed         = "account.closed"

	// Loan origination events
	LoanApplicationSubmitted = "loan.application.submitted"
	LoanApplicationApproved  = "loan.application.approved"
	LoanApplicationRejected  = "loan.application.rejected"
	LoanDocumentsVerified    = "loan.documents.verified"
	LoanCreditAssessed       = "loan.credit.assessed"

	// Loan management events
	LoanDisbursed         = "loan.disbursed"
	LoanRepaymentReceived = "loan.repayment.received"
	LoanDPDUpdated        = "loan.dpd.updated"
	LoanStageChanged      = "loan.stage.changed"
	LoanClosed            = "loan.closed"
	LoanWrittenOff        = "loan.written.off"
	LoanModified          = "loan.modified"

	// Payment events
	PaymentInitiated = "payment.initiated"
	PaymentCompleted = "payment.completed"
	PaymentFailed    = "payment.failed"
	PaymentReversed  = "payment.reversed"

	// Float events
	FloatDrawn             = "float.drawn"
	FloatRepaid            = "float.repaid"
	FloatFeeCharged        = "float.fee.charged"
	FloatRestrictionApplied = "float.restriction.applied"
	FloatLimitChanged      = "float.limit.changed"

	// AML / compliance events
	AMLAlertRaised   = "aml.alert.raised"
	AMLSARFiled      = "aml.sar.filed"
	CustomerKYCPassed = "customer.kyc.passed"
	CustomerKYCFailed = "customer.kyc.failed"

	// Customer events
	CustomerCreated = "customer.created"
	CustomerUpdated = "customer.updated"

	// Fund transfer events
	TransferInitiated = "transfer.initiated"
	TransferCompleted = "transfer.completed"
	TransferFailed    = "transfer.failed"

	// Mobile wallet events
	MobileUserRegistered    = "mobile.user.registered"
	MobileTransferCompleted = "mobile.transfer.completed"
	MobileTransferFailed    = "mobile.transfer.failed"

	// Bill pay events
	BillPaymentCompleted = "bill.payment.completed"
	BillPaymentFailed    = "bill.payment.failed"

	// Savings events
	SavingsGoalCreated      = "savings.goal.created"
	SavingsDeposit          = "savings.deposit"
	SavingsWithdrawal       = "savings.withdrawal"
	SavingsAutoSaveExecuted = "savings.auto.save.executed"

	// Shop / BNPL events
	ShopOrderPlaced   = "shop.order.placed"
	ShopOrderShipped  = "shop.order.shipped"
	ShopOrderDelivered = "shop.order.delivered"
	ShopBNPLApproved  = "shop.bnpl.approved"

	// Fraud detection events
	FraudAlertRaised  = "fraud.alert.raised"
	FraudBlockAccount = "fraud.block.account"

	// Overdraft events
	OverdraftApplied          = "overdraft.applied"
	OverdraftDrawn            = "overdraft.drawn"
	OverdraftRepaid           = "overdraft.repaid"
	OverdraftSuspended        = "overdraft.suspended"
	OverdraftInterestCharged  = "overdraft.interest.charged"
	OverdraftFeeCharged       = "overdraft.fee.charged"
	OverdraftDPDUpdated       = "overdraft.dpd.updated"
	OverdraftStageChanged     = "overdraft.stage.changed"
	OverdraftBillingStatement = "overdraft.billing.statement"
)
